crates/xilem_html/src/context.rs

```diff
@@ -1,17 +1,51 @@
 use std::any::Any;
 
 use bitflags::bitflags;
-use wasm_bindgen::JsCast;
+use wasm_bindgen::{JsCast, UnwrapThrowExt};
 use web_sys::Document;
 
 use xilem_core::{Id, IdPath};
 
-use crate::{app::AppRunner, Message, HTML_NS, SVG_NS};
+use crate::{
+    app::AppRunner,
+    diff::{diff_kv_iterables, Diff},
+    vecmap::VecMap,
+    AttributeValue, Message,
+};
+
+type CowStr = std::borrow::Cow<'static, str>;
+
+fn set_attribute(element: &web_sys::Element, name: &str, value: &str) {
+    // we have to special-case `value` because setting the value using `set_attribute`
+    // doesn't work after the value has been changed.
+    if name == "value" {
+        let element: &web_sys::HtmlInputElement = element.dyn_ref().unwrap_throw();
+        element.set_value(value)
+    } else if name == "checked" {
+        let element: &web_sys::HtmlInputElement = element.dyn_ref().unwrap_throw();
```

Comment:
This is a bit surprising

Reply:
Ok, at least for `value` there indeed seems to be an issue with clearing the value, it just isn't cleared in the input.

I think this is a good motivation to properly support DOM attributes instead of just HTML attributes.